core/ecschema-metadata/src/UnitQuery/UnitQuery.ts

```diff
@@ -0,0 +1,171 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { BentleyError, BentleyStatus } from "@bentley/bentleyjs-core";
+import { SchemaContext, SchemaItem, SchemaItemKey, SchemaKey, Unit } from "../ecschema-metadata";
+import { SchemaItemType } from "../ECObjects";
+import { UNIT_DATA } from "./UnitsData";
```

Comment:
Why are we hardcoding this Units data into ecschema-metadata? If anything this would need to always be pulled from the context rather than hardcoded....

Reply:
Changed to take extra data associated with Units in constructor